lib/pushy_client.rb

```diff
@@ -224,10 +224,31 @@ def get_config
     resource = "/pushy/config/#{node_name}"
 
     Chef::Log.info "[#{node_name}] Retrieving configuration from #{chef_server_url}/#{resource}: ..."
-    esc_key = CGI::escape(@client_curve_pub_key)
+    esc_key = CGI.escape(@client_curve_pub_key)
     version = PushyClient::PROTOCOL_VERSION
     resource = "pushy/config/#{node_name}?ccpk=#{esc_key}&version=#{version}"
 
+    Chef::Config[:rest_timeout] = 5
+
+    if @config.nil?
+      # We are being called from the `start` method not `trigger_reconfigure`
+      # so we know this is the first time fetching the config after a clean service start/restart.
+      # We therefore want to give ample opportunity to fetch the config, otherwise
+      # we raise and exception and the service stops - hopefully to be started again
```

Comment:
This all seems reasonable. As a note, we should make sure that the 'started again on next CCR' works as intended when we're running push client as a windows service. This isn't intended to be a blocker for this PR; as it doesn't make any problems worse AFAIK.

Reply:
Thanks @markan I'll track this very scenario and create an issue if it's an item to address.